src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -33,7 +33,16 @@ namespace Dynamo.ViewModels
 {
     public interface IDynamoViewModel : INotifyPropertyChanged
     {
-        ObservableCollection<WorkspaceViewModel> Workspaces { get; set; } 
+        ObservableCollection<WorkspaceViewModel> Workspaces { get; set; }
+
+        /// <summary>
+        /// Executes node and connector creation commands to create node1 and connect it to node2
+        /// </summary>
+        /// <param name="node1">input node to create</param>
+        /// <param name="node2">node to connect to</param>
+        /// <param name="portIndex1">output port of input node</param>
+        /// <param name="portIndex2">input node of given node</param>
+        void CreateAndConnectInputNode(NodeModel node1, NodeModel node2, int portIndex1, int portIndex2);
```

Comment:
I see @pboyer's concern. We could not break `CreateAndConnectInputNode` into `CreateNode` and `CreateConnector` because that would end up being two different actions on the undo stack (i.e. requiring double-undo to completely undo these actions). Since we're exposing `ICommandExecutive`, then we will introduce a new `RecordableCommand` to perform these two actions in one go: create a new node, and make connection. Does that make sense?